File: C++/FindTheUniqueNumber/FindTheUniqueNumber/FindTheUniqueNumber.cpp
#include <iostream>
#include <algorithm>
#include <vector>
using namespace std;

float find_uniq(const vector<float>& v)
{
    if ((v[0] != v[1]) && (v[0] != v[2]))return v[0];
    if (v[v.size()-1] != v[v.size() - 2] && v[v.size()-1] != v[v.size() - 3])return v[v.size()-1];

    for (int i = 1; i < int(v.size()) - 1; i++)
        if (v[i] != v[i - 1] && v[i] != v[i + 1])return v[i];
   
    return -1;
}

bool equal(float a, float b) {
    return a == b;
}

int main()
{
    cout << std::boolalpha;

    cout << equal(find_uniq({ 3,2,2 }), 3.f) << "\n";
    cout << equal(find_uniq({ 1,2,1 }), 2.f) << "\n";
    cout << equal(find_uniq({ 1, 1, 1, 2, 1, 1 }),2.f)<< "\n";
    cout << equal(find_uniq({ 0, 0, 0.55, 0, 0 }), 0.55f) << "\n";
    cout << equal(find_uniq({999.666,  999.666, 999.666, 999.666, 999.666, 999.666, 999.666, 999, 999.666}), 999.f) << "\n";

	return 0;
}

File: Python/findTheUniqueNumber.py
"""
There is an array with some numbers. All numbers are equal except for one. Try to find it!

find_uniq([ 1, 1, 1, 2, 1, 1 ]) == 2
find_uniq([ 0, 0, 0.55, 0, 0 ]) == 0.55
It’s guaranteed that array contains at least 3 numbers.

The tests contain some very huge arrays, so think about performance.

This is the first kata in series:

Find the unique number (this kata)
Find the unique string
Find The Unique
"""

def find_uniq(arr):
    tmp1 = arr[0]
    counter = 0
    converted = set(arr)
    for n in range(3):
        if arr[n] == tmp1:
            counter += 1
    if counter == 1: return tmp1
    for n in converted:
        if n != tmp1:
            return n

print(find_uniq([0,0,0,1,0,0,0]))